### DISCUSSION TITLE
Using Podman on Linux workstations, especially Fedora 35

### DISCUSSION THREAD

**Discussion by PavelSosin-320** [containers/podman#12693] [General]
Currently, Podman is delivered with Fedora 35 Workstation but doesn't work correctly. Unlike Server and "Regular" distros without Desktop Podman has very few dependencies - it needs only systemd manager for control groups handling and HOME, XDG_DATA_HOME and XDG_RUNTIME_DIR mounted at the System boot and user's login. When Podman is run on Linux workstation inside Desktop session it becomes very dependent from many system or user services like systemd-homed, dbus-broker, journald,  that provide necessary services to the Podman itself, Podman service and Podman containers running as Systemd units. Analyzing Systemd units tree I found that other applications don't hesitate to check, wait and trigger starting of all used services: those that use storage waits mount and home.mount, those that use journal wait for journald, those that manages control groups wait for session service, those that need networking wait for network.target. those that communicate with other services wait for dbus-broker, etc.
It looks like that they use transient systemd unit as a dependencies helper. etc. Podman doesn't do it and it causes synchronization with systemd-homed problems. If any of dependency fails to start the Podman behavior turns to be unpredictable.

**Comment by PavelSosin-320** [comment]
Half year later: Podman 4 doesn't work on Fedora 36 Workstation with Gnome 4.2 doesn't work at all: podman start of existing container fails  when container was created by rootless user and the same user attempts to start it. Very strange set of options is passed to conmon and then crun does very strange manipulation with /proc/self/oom_adj that fails. See log below:
  [pavelsosin@Dell ~]$ podman container create -i --log-level=debug --log-driver=journald --name=fedora registry.fedoraproject.org/fedora
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called create.PersistentPreRunE(podman container create -i --log-level=debug --log-driver=journald --name=fedora registry.fedoraproject.org/fedora) 
DEBU[0000] Merged system config "/usr/share/containers/containers.conf" 
DEBU[0000] Merged system config "/home/pavelsosin/.config/containers/containers.conf" 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /home/pavelsosin/.local/share/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver btrfs                     
DEBU[0000] Using graph root /home/pavelsosin/.local/share/containers/storage 
DEBU[0000] Using run root /run/user/1000/containers/storage 
DEBU[0000] Using static dir /home/pavelsosin/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/user/1000/libpod/tmp      
DEBU[0000] Using volume path /home/pavelsosin/.local/share/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "btrfs" 
DEBU[0000] Initializing event backend journald          
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/crun"            
INFO[0000] Setting parallel job count to 25             
DEBU[0000] Pulling image registry.fedoraproject.org/fedora (policy: missing) 
DEBU[0000] Looking up image "registry.fedoraproject.org/fedora" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:latest" ... 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage ([btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5) 
DEBU[0000] Looking up image "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:latest" ... 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Found image "registry.fedoraproject.org/fedora:latest" as "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Found image "registry.fedoraproject.org/fedora:latest" as "registry.fedoraproject.org/fedora:latest" in local containers storage ([btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5) 
DEBU[0000] Looking up image "registry.fedoraproject.org/fedora" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:latest" ... 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage ([btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5) 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] using systemd mode: false                    
DEBU[0000] setting container name fedora                
DEBU[0000] No hostname set; container's hostname will default to runtime default 
DEBU[0000] Loading seccomp profile from "/usr/share/containers/seccomp.json" 
DEBU[0000] Allocated lock 40 for container 578d79d46b74559da62bc9674b5f9639d242e957e10ca6bee1d16c017dab464e 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Failed to create container fedora(578d79d46b74559da62bc9674b5f9639d242e957e10ca6bee1d16c017dab464e): the container name "fedora" is already in use by "1ad591fa51be825adc648149ed6021cce7de0a91643ac8d9417be79b5c2aaa73". You have to remove that container to be able to reuse that name.: that name is already in use 
Error: error creating container storage: the container name "fedora" is already in use by "1ad591fa51be825adc648149ed6021cce7de0a91643ac8d9417be79b5c2aaa73". You have to remove that container to be able to reuse that name.: that name is already in use
[pavelsosin@Dell ~]$ podman container create -i --log-level=debug --log-driver=journald --name=fedora36 registry.fedoraproject.org/fedora
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called create.PersistentPreRunE(podman container create -i --log-level=debug --log-driver=journald --name=fedora36 registry.fedoraproject.org/fedora) 
DEBU[0000] Merged system config "/usr/share/containers/containers.conf" 
DEBU[0000] Merged system config "/home/pavelsosin/.config/containers/containers.conf" 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /home/pavelsosin/.local/share/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver btrfs                     
DEBU[0000] Using graph root /home/pavelsosin/.local/share/containers/storage 
DEBU[0000] Using run root /run/user/1000/containers/storage 
DEBU[0000] Using static dir /home/pavelsosin/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/user/1000/libpod/tmp      
DEBU[0000] Using volume path /home/pavelsosin/.local/share/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "btrfs" 
DEBU[0000] Initializing event backend journald          
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/crun"            
INFO[0000] Setting parallel job count to 25             
DEBU[0000] Pulling image registry.fedoraproject.org/fedora (policy: missing) 
DEBU[0000] Looking up image "registry.fedoraproject.org/fedora" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:latest" ... 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage ([btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5) 
DEBU[0000] Looking up image "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:latest" ... 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Found image "registry.fedoraproject.org/fedora:latest" as "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Found image "registry.fedoraproject.org/fedora:latest" as "registry.fedoraproject.org/fedora:latest" in local containers storage ([btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5) 
DEBU[0000] Looking up image "registry.fedoraproject.org/fedora" in local containers storage 
DEBU[0000] Normalized platform linux/amd64 to {amd64 linux  [] } 
DEBU[0000] Trying "registry.fedoraproject.org/fedora:latest" ... 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage 
DEBU[0000] Found image "registry.fedoraproject.org/fedora" as "registry.fedoraproject.org/fedora:latest" in local containers storage ([btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5) 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] Inspecting image 3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5 
DEBU[0000] using systemd mode: false                    
DEBU[0000] setting container name fedora36              
DEBU[0000] No hostname set; container's hostname will default to runtime default 
DEBU[0000] Loading seccomp profile from "/usr/share/containers/seccomp.json" 
DEBU[0000] Allocated lock 40 for container 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f 
DEBU[0000] parsed reference into "[btrfs@/home/pavelsosin/.local/share/containers/storage+/run/user/1000/containers/storage]@3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] exporting opaque data as blob "sha256:3a66698e604003f7822a0c73e9da50e090fda9a99fe1f2e1e2e7fe796cc803d5" 
DEBU[0000] Created container "3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f" 
DEBU[0000] Container "3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f" has work directory "/home/pavelsosin/.local/share/containers/storage/btrfs-containers/3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f/userdata" 
DEBU[0000] Container "3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f" has run directory "/run/user/1000/containers/storage/btrfs-containers/3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f/userdata" 
3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f
DEBU[0000] Called create.PersistentPostRunE(podman container create -i --log-level=debug --log-driver=journald --name=fedora36 registry.fedoraproject.org/fedora) 
[pavelsosin@Dell ~]$ 

================================
[pavelsosin@Dell ~]$ podman container start -i --log-level=debug --log-driver=journald -l
Error: unknown flag: --log-driver
See 'podman container start --help'
[pavelsosin@Dell ~]$ podman container start -i --log-level=debug -l
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called start.PersistentPreRunE(podman container start -i --log-level=debug -l) 
DEBU[0000] Merged system config "/usr/share/containers/containers.conf" 
DEBU[0000] Merged system config "/home/pavelsosin/.config/containers/containers.conf" 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /home/pavelsosin/.local/share/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver btrfs                     
DEBU[0000] Using graph root /home/pavelsosin/.local/share/containers/storage 
DEBU[0000] Using run root /run/user/1000/containers/storage 
DEBU[0000] Using static dir /home/pavelsosin/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/user/1000/libpod/tmp      
DEBU[0000] Using volume path /home/pavelsosin/.local/share/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "btrfs" 
DEBU[0000] Initializing event backend journald          
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/crun"            
INFO[0000] Setting parallel job count to 25             
DEBU[0000] Made network namespace at /run/user/1000/netns/netns-c2aeee07-439b-8756-0566-1eaabd245230 for container 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f 
DEBU[0000] [graphdriver] trying provided driver "btrfs" 
DEBU[0000] Mounted container "3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f" at "/home/pavelsosin/.local/share/containers/storage/btrfs/subvolumes/96bc3871dcc6df39c088e8054befbe9aeb17ad060ed3f4534b74402ff2dfdaa3" 
DEBU[0000] Created root filesystem for container 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f at /home/pavelsosin/.local/share/containers/storage/btrfs/subvolumes/96bc3871dcc6df39c088e8054befbe9aeb17ad060ed3f4534b74402ff2dfdaa3 
DEBU[0000] slirp4netns command: /usr/bin/slirp4netns --disable-host-loopback --mtu=65520 --enable-sandbox --enable-seccomp --enable-ipv6 -c -e 3 -r 4 --netns-type=path /run/user/1000/netns/netns-c2aeee07-439b-8756-0566-1eaabd245230 tap0 
DEBU[0000] /etc/system-fips does not exist on host, not mounting FIPS mode subscription 
DEBU[0000] Setting Cgroups for container 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f to user.slice:libpod:3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f 
DEBU[0000] reading hooks from /usr/share/containers/oci/hooks.d 
DEBU[0000] Workdir "/" resolved to host path "/home/pavelsosin/.local/share/containers/storage/btrfs/subvolumes/96bc3871dcc6df39c088e8054befbe9aeb17ad060ed3f4534b74402ff2dfdaa3" 
DEBU[0000] Created OCI spec for container 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f at /home/pavelsosin/.local/share/containers/storage/btrfs-containers/3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f/userdata/config.json 
**DEBU[0000] /usr/bin/conmon messages will be logged to syslog** 
DEBU[0000] running conmon: /usr/bin/conmon               args="[--api-version 1 -c 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f -u 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f -r /usr/bin/crun -b /home/pavelsosin/.local/share/containers/storage/btrfs-containers/3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f/userdata -p /run/user/1000/containers/storage/btrfs-containers/3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f/userdata/pidfile -n fedora36 --exit-dir /run/user/1000/libpod/tmp/exits --full-attach -s -l journald --log-level debug --syslog -i --conmon-pidfile /run/user/1000/containers/storage/btrfs-containers/3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f/userdata/conmon.pid --exit-command /usr/bin/podman --exit-command-arg --root --exit-command-arg /home/pavelsosin/.local/share/containers/storage --exit-command-arg --runroot --exit-command-arg /run/user/1000/containers/storage --exit-command-arg --log-level --exit-command-arg debug --exit-command-arg --cgroup-manager --exit-command-arg systemd --exit-command-arg --tmpdir --exit-command-arg /run/user/1000/libpod/tmp --exit-command-arg --network-config-dir --exit-command-arg  --exit-command-arg --network-backend --exit-command-arg cni --exit-command-arg --volumepath --exit-command-arg /home/pavelsosin/.local/share/containers/storage/volumes --exit-command-arg --runtime --exit-command-arg crun --exit-command-arg --storage-driver --exit-command-arg btrfs --exit-command-arg --events-backend --exit-command-arg journald --exit-command-arg --syslog --exit-command-arg container --exit-command-arg cleanup --exit-command-arg 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f]"
INFO[0000] Running conmon under slice user.slice and unitName libpod-conmon-3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f.scope 
[conmon:d]: failed to write to /proc/self/oom_score_adj: Permission denied

DEBU[0000] Received: -1                                 
DEBU[0000] Cleaning up container 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f 
DEBU[0000] Tearing down network namespace at /run/user/1000/netns/netns-c2aeee07-439b-8756-0566-1eaabd245230 for container 3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f 
DEBU[0000] Unmounted container "3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f" 
**Error: unable to start container "3b1fa1e7c9a922684d52cbae0d47fc451bb90ffa469220e307ec3ded69622a8f": crun: [conmon:d]: failed to **write to /proc/self/oom_score_adj: Permission denied****

**open executable: Permission denied: OCI permission denied**
[pavelsosin@Dell ~]$ 

Why conmon is started with syslog option? In the workstation environment it contradicts the log separation requirement and dangerous from the security point of view to use syslog by the nonprivileged rootless user. Session-level journald socket looks more safe. 
Why crun tries to adjust oom limit if Fedora has its own service to limit session resources. Conmon can preserve resources using cgroups mechanism inside session's resources boundaries. systemd-oomd and its configuration can care about memory usage in the batter way than Conmon. Maybe, Conmon is too limited to be really useful in the Workstation where user can create many sessions and gnome-terminal can have many tabs. Different container runtimes can use system resources in very different way, what Conmon can say me if I use Kata runtime?

**Reply by vrothberg** [reply]
What is your concrete setup?  I think it would be much easier for us to know more about your setup.  It's very time-consuming to parse the wall of text.

**Comment by PavelSosin-320** [comment]
@vrothberg Maybe, [BPF](https://github.com/containers/oci-seccomp-bpf-hook) related? Indeed, it was missed in my setup.
This is virgin Fedora 35 WS installed from USB on the Dell laptop and progressed to the 36, then all updates were installed, including Podman updates. Currently:
podman version
Client:       Podman Engine
Version:      4.1.0
API Version:  4.1.0
Go Version:   go1.18
Built:        Fri May  6 19:15:54 2022
OS/Arch:      linux/amd64
conmon --version
conmon version 2.1.0
commit: 
crun --version
crun version 1.4.5
commit: c381048530aa750495cf502ddb7181f2ded5b400
spec: 1.0.0
+SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL

[pavelsosin@Dell ~]$ podman info
host:
  arch: amd64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 77.9
    systemPercent: 4
    userPercent: 18.11
  cpus: 8
  distribution:
    distribution: fedora
    variant: workstation
    version: "36"
  eventLogger: journald
  hostname: Dell
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.17.12-300.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 448643072
  memTotal: 8182312960
  networkBackend: cni
  ociRuntime:
    name: crun
    package: crun-1.4.5-1.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.5
      commit: c381048530aa750495cf502ddb7181f2ded5b400
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    **rootless: true**
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      **libseccomp: 2.5.3**
  swapFree: 6586302464
  swapTotal: 8182034432
  uptime: 62h 11m 40.05s (Approximately 2.58 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/pavelsosin/.config/containers/storage.conf
  containerStore:
    number: 23
    paused: 0
    running: 0
    stopped: 23
  graphDriverName: btrfs
  graphOptions: {}
  graphRoot: /home/pavelsosin/.local/share/containers/storage
  graphRootAllocated: 998500204544
  graphRootUsed: 19166232576
  graphStatus:
    Build Version: Btrfs v5.16.2
    Library Version: "102"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 11
  runRoot: /run/user/1000/containers/storage
  volumePath: /home/pavelsosin/.local/share/containers/storage/volumes
version:
  APIVersion: 4.1.0
  Built: 1651853754
  BuiltTime: Fri May  6 19:15:54 2022
  GitCommit: ""
  GoVersion: go1.18
  Os: linux
  OsArch: linux/amd64
  Version: 4.1.0

==================
SELinux monitor doesn't jump when the error occurred.
All tested Images are standard images from DockerHub
====================================================
Bash process capabilities are:
capsh --print
Current: =
Bounding set =cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read,cap_perfmon,cap_bpf,cap_checkpoint_restore
Ambient set =
Current IAB: 
Securebits: 00/0x0/1'b0 (no-new-privs=0)
 secure-noroot: no (unlocked)
 secure-no-suid-fixup: no (unlocked)
 secure-keep-caps: no (unlocked)
 secure-no-ambient-raise: no (unlocked)
uid=1000(pavelsosin) euid=1000(pavelsosin)
gid=1000(pavelsosin)
groups=10(wheel),1000(pavelsosin)
Guessed mode: UNCERTAIN (0)
=========================================
uname -a
**Linux Dell 5.17.12-300.fc36.x86_64 #1 SMP PREEMPT Mon May 30** 16:56:53 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**Comment by PavelSosin-320** [comment]
I suppose that the root cause is located in the OS.exec which Conmon uses to run crun. This is omething strange with OS.exec in thr golang libraries for all platforms because Google search returns dozens of reported issues regarding os.fork and os.exec that return "Permition denied" or "can't access". It simply doesn't work. Maybe, because it needs shell to run and any shell I tried to fork from the  gnome-terminal can't execute any executable, bash is fails for example. I would  be happy to provide output of bash capsh --print but it fails with the same error - '/usr/bin/capsh couldn't execute binary file". when executed by either root or rootless user. 
There is essential difference between bash running in the server-like environment and bash running in the Gnome because in the last case  bash has parent started by systemd unit and inherits privileges via gnome-session, gnome-terminal, etc. This unit supplies its own bounding set. It can be very limited.

**Comment by PavelSosin-320** [comment]
@vrothberg Only for curiosity: running Fedora image inside Podman machine on Fedora ends with the same error:
Running conmon....
[conmon:d] DEBUG? failed to write to /proc/self/oon/score_adj: permission denied
Received 2123
.....
Then starting and starting container successfully 
and then
Received shutdown.Stop, terminating!
And something kills container without any SELinux or audit record. Maybe, conmon propagates received SIGKILL to container.

**Comment by PavelSosin-320** [comment]
Inside Podman machine from Fedora image it works when cgroups is set to disabled. But there podman runs inside plane remote session as core user. I think that the conclusion is that in the Workstation environment either podman or conmon creates container's cgroup incorrectly in the session hierarchy and the container is stillborn. Indeed, this is discussable what is container/conmon cgroup in the Gnome session. Could somebody pay attention?

**Comment by PavelSosin-320** [comment]
@mheon I would like to try running podman service inside Fedora CoreOS VM under **Fedora WS Boxes**. Unfortunately, CoreOS ISO image is published without dnf. So, I need url of the good Podman release candidate rpm. From this point everything should be very similar to any other Workstation running podman via podman machine create command or podman-in-docker container.

**Reply by rhatdan** [reply]
Podman should be available in Fedora CoreOS?

**Reply by PavelSosin-320** [reply]
According to [Fedora CoreOS](https://docs.fedoraproject.org/en-US/fedora-coreos/) purpose I would expect that it has to be available. CoreOS itself is available out-of-the-box in Fedora workstation including one touch start via Boxes.

**Comment by PavelSosin-320** [comment]
Yes!!!! After the recent Fedora 36 full upgrade I got podman 4.1 inside Fedora CoreOS Gnome Box  including podman system service. In other words Podman service can run virtualized on the Fedora 36 Workstation at least at POC level.
In other words, The Podman for Windows architecture can be replicated to the Linux workstation but with much better hardware and storage access quality.
